### DISCUSSION TITLE
How to create an agent during a simulation

### DISCUSSION THREAD

**Discussion by notionparallax** [JoelForamitti/agentpy#42] [Q&A]
I have a simple simulation of a company. It is doing a great job of modelling promotions, but I'd also like to be able to model people joining and leaving the company.

Leaving is easy, I can just set a flag to inactive or something, but joining is more complicated. Do I need to make a bunch of agents during setup and set their state to as yet unknown, or can I create an agent during a step and add them in?

The person class is defined like this:

```
class PersonAgent(ap.Agent):

    def setup(self):
        p = PEOPLE_DF.iloc[self.id] # 👈 the existing people are in a spreadsheet
        self.name = p.full_name
        self.gender = get_gender(p.gender)
        self.bvn_rank = get_rank(p.b_rank)
        # self.capability = float(p.capability)
        print()

    def rank_transition(self):
        self.bvn_rank = transition(self.b_rank, self.gender)
```

I'm guessing I'd do something with the __init__, but I've had no luck figuring that out.

This seems similar to #20, but I'm not trying to make different types of agents, but add more of a single type of agent. 

I [posted this on stack overflow too](https://stackoverflow.com/questions/72358308/using-agentpy-how-to-create-an-agent-during-a-simulation), before I found this forum.

**Comment by jofmi** [comment]
Hi @notionparallax, I responded on stack overflow :) https://stackoverflow.com/questions/72358308/using-agentpy-how-to-create-an-agent-during-a-simulation/72374989#72374989
